### DISCUSSION TITLE
commercial use

### DISCUSSION THREAD

**Discussion by tobehn** [Budibase/budibase#2725] [General]
Hi Budibase Team,
i know, it might be a little bit early. 

I have a customer who has just asked for a small application to track and document abnormalities, findings, changes of parameters, malfunctions and failures in a monitoring system i have set up for him.

The Monitoring System basically is Grafana, InfluxDB, NodRED and some neat Edgedevices to gain sensor data. Its all docker based and run with traefik as reverse proxy and loadbalancer on an ubuntu machine.

I would really love to try to realize this with budibase as from my experience so far it looks like a perfect fit. 

Demands:
- up to 10 Users (later maybe scaled up) can input data like despcriptions of failures parameter changes etc.
- the inputs of each user cannot be edited by the other users
- single sign on with Grafana ( OpenID no experience on my side yet)
- perhaps some (email) notifications when entries are saved, marked closed, overdue or something like that

Its a pretty data sensitive customer and i do not want to intergrate to much of external services for automation and stuff.
I would definatley want to host it next to the traefik setup.
I could earn money for my programming hours with this and i would want to push something over to BB

First of, do you think this task can easily be done with budibase (i also might come with individual requests).

- Can i use your system for commercial use as of today
- How can i make sure to give something back and what are usually your ideas about that?

I am looking forward hearing from you.

**Comment by mjashanks** [comment]
Hey @tobehn.

> the inputs of each user cannot be edited by the other user

This can be done from a user interface perspective. You can make a screen that only allows users to edit their own items. However, the backend APIs would still be open to anyone with "Write" access to the table/query

> single sign on with Grafana

Is Grafana an auth provider of its own? It looks to me like they link to other SSO providers. In which case, Budibase would link to the same (?)

> perhaps some (email) notifications when entries are saved, marked closed, overdue or something like that

This is possible today with automations. Automations are about to get a lot better too!

> Can I use your system for commercial use as of today

Yes - please do! Open-source and free to self-host.

> How can i make sure to give something back and what are usually your ideas about that?

Let's talk! The way we view monetization:

1. Our future cloud platform 
2. Priority support contracts
3. Future enterprise features for self-hosted. These will be targeted at managing large deployments, so they probably won't be relevant to 10 user instances.

The best things that you can do for us right now is

- Continue to be an active community member. Feel free to contribute in any way, whether that's coding integrations, fixing docs, answering questions or just giving feedback
- Shout about us if you like Budibase!

Happy to connect anytime - book a call with me: https://calendly.com/mjashanks/budibase !
